python/python/tests/torch_tests/test_distance.py

```diff
@@ -43,12 +52,52 @@ def test_l2_distance():
     x = np.random.randn(20, 256).astype(np.float32)
     y = np.random.rand(100, 256).astype(np.float32)
 
-    dist = l2_distance(torch.from_numpy(x), torch.from_numpy(y))
-    assert dist.shape == (20, 100)
+    part_ids, dist = l2_distance(torch.from_numpy(x), torch.from_numpy(y))
+    assert dist.shape == (20,)
 
     expect_arr = (
         np.array([[np.linalg.norm(x_row - y_row) ** 2 for y_row in y] for x_row in x])
         .astype(np.float32)
         .reshape(20, 100)
     )
-    assert np.allclose(dist.cpu(), expect_arr)
+    expect_part_ids = np.argmin(expect_arr, axis=1)
+    expect_dists = np.take_along_axis(
+        expect_arr, expect_part_ids.reshape((-1, 1)), axis=1
+    ).reshape(-1)
+    print(expect_dists, dist)
+    assert np.allclose(dist.cpu(), expect_dists)
+    assert np.allclose(part_ids.cpu(), expect_part_ids)
+
+
+@pytest.mark.skip()
+def test_large_cosine_distance():
+    """Test CUDA Out of memory"""
+    from lance.torch.distance import cosine_distance
+
+    rng = np.random.default_rng()
+
+    x = rng.random((1024 * 40, 1536), dtype="float32")
+    y = rng.random((1024 * 64, 1536), dtype="float32")
+
+    (part_ids, dist) = cosine_distance(
+        torch.from_numpy(x).to("cuda"), torch.from_numpy(y).to("cuda")
+    )
+    assert part_ids.shape == (1024 * 40,)
+    assert dist.shape == (1024 * 40,)
+
+
+@pytest.mark.skip()
+def test_large_l2_distance():
+    """Test CUDA Out of memory"""
+    from lance.torch.distance import l2_distance
+
+    rng = np.random.default_rng()
+
+    x = rng.random((1024 * 40, 1536), dtype="float32")
+    y = rng.random((1024 * 64, 1536), dtype="float32")
+
+    (part_ids, dist) = l2_distance(
+        torch.from_numpy(x).to("cuda"), torch.from_numpy(y).to("cuda")
+    )
+    assert part_ids.shape == (1024 * 40,)
+    assert dist.shape == (1024 * 40,)
```

Comment:
Maybe an explanation as to how these trigger an out of memory bug?  Is it just because that's so much data it is guaranteed to overflow the device regardless of the device?

Reply:
For `[N, D]` vectors to test against to `[M, D]` centroids,  it creates intermedia `[N, M]` matrix that has pair wise distances between the two. The fix in this PR is reduce `[N, M]` to multiple of `[split_size, M]` , and by controling `split_size`, it controls the memory used in GPU.

Reply:
Added to the test docstring as well.